lib/taskana-core/src/main/java/pro/taskana/ClassificationService.java

```diff
@@ -44,6 +46,34 @@
      */
     Classification getClassification(String key, String domain) throws ClassificationNotFoundException;
 
+    /**
+     * Delete a non-root-classification.
+     *
+     * @param classificationKey
+     *              the key of the classification you want to delete.
+     * @param domains
+     *              the domains for which you want to delete the classification.
+     * @throws ClassificationInUseException
+     *              if there are Task existing, which refer to this classification.
+     * @throws NotAuthorizedException
+     *              if you're not authorized to delete the classification.
+     * @throws ClassificationNotFoundException
+     *              if for an domain no classification specification is found.
+     */
+    void deleteClassification(String classificationKey, String... domains) throws ClassificationInUseException, NotAuthorizedException, ClassificationNotFoundException;
```

Comment:
NotAuthorizedException shouldn´t be thrown here. It´s a Exception for Workbasket-Permissions and not for changing classifications. 
I guess you can ignore this case, because who ever will be able to see a data in a workbasket with a classification got at least READ permission.

**Please ask @holgerhagen if there is a special permission for this interaction or if you can remove this throwing argument.**

Reply:
Delete classification will need a role based authorization. But we don't have it right know. So Marcel is right. NotAuthorized should not happen.